web/web.go

```diff
@@ -0,0 +1,773 @@
+			}
+		}
+		summary[t.SrcTable] = body.String()
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(summary)
+}
+
+// getOverview returns the overview of conversion.
+func getOverview(w http.ResponseWriter, r *http.Request) {
+	var buf bytes.Buffer
+	bufWriter := bufio.NewWriter(&buf)
+	internal.GenerateReport(app.driver, app.conv, bufWriter, nil, false, false)
+	bufWriter.Flush()
+	overview := buf.String()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(overview)
+}
+
+// getTypeMap returns the source to Spanner typemap only for the
+// source types used in current conversion.
+func getTypeMap(w http.ResponseWriter, r *http.Request) {
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	var typeMap map[string][]typeIssue
+	switch app.driver {
+	case "mysql", "mysqldump":
+		typeMap = mysqlTypeMap
+	case "postgres", "pg_dump":
+		typeMap = postgresTypeMap
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	// Filter typeMap so it contains just the types SrcSchema uses.
+	filteredTypeMap := make(map[string][]typeIssue)
+	for _, srcTable := range app.conv.SrcSchema {
+		for _, colDef := range srcTable.ColDefs {
+			if _, ok := filteredTypeMap[colDef.Type.Name]; ok {
+				continue
+			}
+			filteredTypeMap[colDef.Type.Name] = typeMap[colDef.Type.Name]
+		}
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(filteredTypeMap)
+}
+
+// setTypeMapGlobal allows to change spanner type globally.
+// It takes a map from source type to spanner type and updates
+// the spanner schema accordingly.
+func setTypeMapGlobal(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var typeMap map[string]string
+	err = json.Unmarshal(reqBody, &typeMap)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	// Redo source-to-Spanner type mapping using t (the mapping specified in the http request).
+	// We drive this process by iterating over the Spanner schema because we want to preserve all
+	// other customizations that have been performed via the UI (dropping columns, renaming columns
+	// etc). In particular, note that we can't just blindly redo schema conversion (using an appropriate
+	// version of 'toDDL' with the new type mapping).
+	for t, spSchema := range app.conv.SpSchema {
+		for col, _ := range spSchema.ColDefs {
+			sourceTable := app.conv.ToSource[t].Name
+			sourceCol := app.conv.ToSource[t].Cols[col]
+			srcCol := app.conv.SrcSchema[sourceTable].ColDefs[sourceCol]
+			if _, found := typeMap[srcCol.Type.Name]; found {
```

Comment:
Maybe add a comment before this line:

// If the srcCol's type is in the map, then recalculate the Spanner type
// for this column using the map. Otherwise, leave the ColDef for this
// column as is.

Reply:
added comment